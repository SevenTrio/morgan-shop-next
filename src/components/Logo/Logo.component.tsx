import { forwardRef, HTMLAttributes } from 'react';
import { ISpacingProps } from 'types/spacing.types';

import { StyledLogo, StyledLogoIcon } from './Logo.styles';

interface ILogoProps extends ISpacingProps, HTMLAttributes<HTMLDivElement> {
  /**
   * The component used for the root node.
   */
  component?: 'div' | 'a';
}

export const Logo = forwardRef<HTMLDivElement, ILogoProps>(
  ({ component = `div`, ...restProps }, ref) => (
    <StyledLogo as={component} ref={ref} {...restProps}>
      <StyledLogoIcon />
    </StyledLogo>
  ),
);

Logo.displayName = `Logo`;
