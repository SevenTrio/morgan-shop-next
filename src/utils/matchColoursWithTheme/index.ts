import { matchColoursWithTheme } from './matchColoursWithTheme';

export default matchColoursWithTheme;
