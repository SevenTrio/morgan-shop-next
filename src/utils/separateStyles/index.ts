import { separateStyles } from './separateStyles';

export default separateStyles;
