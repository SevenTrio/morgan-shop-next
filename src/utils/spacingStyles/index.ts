import { spacingStyles } from './spacingStyles';

export default spacingStyles;
