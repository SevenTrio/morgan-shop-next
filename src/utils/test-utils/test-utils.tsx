import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';
import { Normalize } from 'styled-normalize';

import { GlobalStyle } from 'styles/global-styles';
import { theme } from 'styles/theme';

const AllTheProviders: FC<{ children: React.ReactNode }> = ({ children }) => (
  <ThemeProvider theme={theme}>
    <Normalize />
    <GlobalStyle />
    {children}
  </ThemeProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
