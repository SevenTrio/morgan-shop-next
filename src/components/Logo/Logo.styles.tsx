import styled from 'styled-components';

import spacingStyles from 'utils/spacingStyles';

import LogoIcon from 'icons/logo.svg';

export const StyledLogo = styled.div`
  ${spacingStyles};

  display: flex;
  align-items: center;
  justify-content: center;
  width: 64px;
  height: 64px;

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.sm}px) {
    width: 48px;
    height: 48px;
  }
`;

export const StyledLogoIcon = styled(LogoIcon)`
  width: 36px;
  height: 36px;

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.sm}px) {
    width: 28px;
    height: 28px;
  }
`;
