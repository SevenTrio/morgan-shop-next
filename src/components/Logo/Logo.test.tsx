import { render } from 'utils/test-utils';

import { Logo } from './Logo.component';

describe(`Logo component`, () => {
  describe(`renders correctly`, () => {
    test(`without props`, () => {
      const { container } = render(<Logo />);
      const tree = container.firstChild;

      expect(tree).toMatchSnapshot();
    });
  });

  describe(`applies correct styles`, () => {
    test(`to wrapper component`, () => {
      const { container } = render(
        <Logo m={2} style={{ display: 'none' }} className="Test" />,
      );
      const tree = container.firstChild as HTMLElement;

      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule(`margin`, `8px`);
      expect(tree?.style.display).toBe('none');
      expect(tree?.className.split(' ').includes('Test')).toBe(true);
    });
  });

  describe(`renders correct tag`, () => {
    test(`with component prop`, () => {
      const { container } = render(<Logo component="a" />);
      const tree = container.firstChild;

      expect(tree).toMatchSnapshot();
      expect(tree?.nodeName).toBe(`A`);
    });
  });
});
