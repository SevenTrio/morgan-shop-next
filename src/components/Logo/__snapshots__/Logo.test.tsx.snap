// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component applies correct styles to wrapper component 1`] = `
.c0 {
  margin: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 64px;
  height: 64px;
}

.c1 {
  width: 36px;
  height: 36px;
}

@media screen and (max-width:600px) {
  .c0 {
    width: 48px;
    height: 48px;
  }
}

@media screen and (max-width:600px) {
  .c1 {
    width: 28px;
    height: 28px;
  }
}

<div
  class="c0 Test"
  style="display: none;"
>
  <svg
    class="c1"
  />
</div>
`;

exports[`Logo component renders correct tag with component prop 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 64px;
  height: 64px;
}

.c1 {
  width: 36px;
  height: 36px;
}

@media screen and (max-width:600px) {
  .c0 {
    width: 48px;
    height: 48px;
  }
}

@media screen and (max-width:600px) {
  .c1 {
    width: 28px;
    height: 28px;
  }
}

<a
  class="c0"
>
  <svg
    class="c1"
  />
</a>
`;

exports[`Logo component renders correctly without props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 64px;
  height: 64px;
}

.c1 {
  width: 36px;
  height: 36px;
}

@media screen and (max-width:600px) {
  .c0 {
    width: 48px;
    height: 48px;
  }
}

@media screen and (max-width:600px) {
  .c1 {
    width: 28px;
    height: 28px;
  }
}

<div
  class="c0"
>
  <svg
    class="c1"
  />
</div>
`;
